org.eclipse.gef/src/org/eclipse/gef/editparts/AbstractEditPart.java

```diff
@@ -844,7 +844,9 @@ protected final void registerAccessibility() {
 	 * this method if they need to register this EditPart in additional ways.
 	 */
 	protected void registerModel() {
-		getViewer().getEditPartRegistry().put(getModel(), this);
+		@SuppressWarnings("unchecked")
+		Map<Object, EditPart> registry = (Map<Object, EditPart>) getViewer().getEditPartRegistry();
```

Comment:
I wonder if it's really a good idea for the `getEditPartRegistry()` to return a map with wildcards. All clients that register their own models now have to do the same cast in their edit parts...

Reply:
Thx for bringing it up. I noticed that also and was not sure. I was either thinking of non wildcard returns or an additional helper `registerEditPart(EditPart ep, Object model)`. What do you think is better?

Reply:
I believe the cases where the registry only contains edit parts of a single type (i.e. where one would cast the map to something other than Map\<Object, EditPart\>) are mostly academical. So I would simply remove the wildcard, which would then hopefully minimize the amount of change required by downstream projects.

Reply:
You have a point here. Will change accordingly.

Reply:
done